src/mainviews/create-job.tsx

```diff
@@ -83,7 +83,14 @@ export function CreateJob(props: ICreateJobProps): JSX.Element {
   // Retrieve the environment list once.
   useEffect(() => {
     const setList = async () => {
-      setEnvironmentList(await api.getRuntimeEnvironments());
+      const envList = await api.getRuntimeEnvironments();
+      setEnvironmentList(envList);
+      if (envList.length === 1) {
+        props.handleModelChange({
+          ...props.model,
```

Comment:
@andrii-i The input file and job name were being reset in your latest commit because the `props` is redefined as a new object every render. So basically when `setList()` is defined, it binds the `props` variable to whatever value it was at initial render. This means that when `setList()` is actually executed, `props` is still bound to the value of `props` at initial render, meaning that its property `props.model` is still the value at initial render, i.e. `emptyCreateJobModel()`.

For now, a temporary band-aid is to make sure that the model contains the job name and input file on initial render (see changes in `index.tsx`). This would fail if we had other effect hooks asynchronously setting the state after initial render. The ideal solution would be to support allow `handleModelChange()` to accept a callback that always provides the latest value of `model`, similar to the alternate callback syntax of `useState()[1]`. https://reactjs.org/docs/hooks-reference.html#functional-updates